src/store.rs

```diff
@@ -232,19 +233,19 @@ pub struct PyHttpContext {
 #[pymethods]
 impl PyHttpContext {
     #[new]
-    fn new(url: String) -> Self {
+    fn new(url: String) -> PyResult<Self> {
         let store = match Url::parse(url.as_str()) {
             Ok(url) => HttpBuilder::new()
                 .with_url(url.origin().ascii_serialization())
                 .build()
-                .unwrap(),
+                .map_err(|e| PyValueError::new_err(format!("Error: {:?}", e.to_string())))?,
             Err(_) => HttpBuilder::new().build().unwrap(),
```

Comment:
Sorry, didn't catch the first round there's another unwrap that we should handle. It looks like you can just move it around to the end of the match statement to cover both.